### PR TITLE
fix: use timestamp passed by request for user message created date

### PR DESCRIPTION
**Pull request by goetzrobin**
we pass the timestamp through until we reach the agent's step function this function calls Message.dict_to_message(), where we pass the timestamp as argument for created_at, which is then persisted into the DB

**Please describe the purpose of this pull request.**
Is it to add a new feature? Is it to fix a bug?

**How to test**
How can we test your PR during review? What commands should we run? What outcomes should we expect?
After merging the dev portal changes and creating a new agent duplicated messages should not show up anymore

**Have you tested this PR?**
Have you tested the latest commit on the PR? If so please provide outputs from your tests.

**Related issues or PRs**
Please link any related GitHub [issues](https://github.com/cpacker/MemGPT/issues) or [PRs](https://github.com/cpacker/MemGPT/pulls).

**Is your PR over 500 lines of code?**
If so, please break up your PR into multiple smaller PRs so that we can review them quickly, or provide justification for its length.

**Additional context**
Add any other context or screenshots about the PR here.

### CODE REVIEW

**Comment by cpacker** [comment]
Pending tests LGTM.

Though I think at a later point we may want to revisit the concept of allowing the user/client to specify the timestamp to be stored server-side since it feels like a security risk. Instead maybe we can return the timestamp (and other necessary metadata) in the response of the `POST` call, which should effectively allow the client (eg chat UI) to get a server-verified timestamp immediately after `POST`.

Thoughts @goetzrobin ?

**Comment by goetzrobin** [comment]
@cpacker I agree. I like that or maybe have the first streamed event return metadata like an id & timestamp? I do think moving forward we definitely want to have this server driven.

File: memgpt/agent_store/db.py [hunk 1]
```diff
@@ -314,19 +314,13 @@ def get_all_cursor(
             # cursor logic: filter records based on before/after ID
             if after:
                 after_value = getattr(self.get(id=after), order_by)
-                if reverse:  # if reverse, then we want to get records that are less than the after_value
```
Comment (cpacker) [inline memgpt/agent_store/db.py:317]: We should probably keep `reverse` in, right? Was this just removed for debugging purposes?

File: memgpt/server/server.py [hunk 2]
```diff
@@ -560,7 +563,7 @@ def user_message(
             elif message.startswith("/"):
                 raise ValueError(f"Invalid input: '{message}'")
 
-            packaged_user_message = system.package_user_message(user_message=message)
+            packaged_user_message = system.package_user_message(user_message=message, time=timestamp.isoformat())
```
Comment (cpacker) [inline memgpt/server/server.py:566]: This needs a catch for the `timestamp is None` case

**Review by cpacker — CHANGES_REQUESTED** [review]

**Comment by cpacker** [comment]
> @cpacker I agree. I like that or maybe have the first streamed event return metadata like an id & timestamp? I do think moving forward we definitely want to have this server driven.

Gotcha. Let's merge in this PR for now, but deprecate the ability to pass in `timestamp` via the API call in a subsequent PR.

File: memgpt/agent_store/db.py (hunk 1 above)
Comment (cpacker) [reply memgpt/agent_store/db.py]: @goetzrobin I reverted the change on this file - I'm assuming that the diff isn't actually needed in this PR?

**Pull request merged**